src/hubbleds/pages/02-distance-introduction/__init__.py

```diff
@@ -1,45 +1,65 @@
-#
-#
-# component_state = ComponentState()
-#
-# @solara.component
-# def Page():
-#     load_custom_vue_components()
-#
-#     Stage2Slideshow(
-#         step = component_state.distance_slideshow_state.step_dist.value,
-#         max_step_completed = component_state.distance_slideshow_state.max_step_completed.value,
-#         event_on_slideshow_finished=lambda *args: component_state.distance_slideshow_state.complete.set(
-#                             True
-#                         ),
-#         length = 13,
-#         titles = [
-#             "1920's Astronomy",
-#             "1920's Astronomy",
-#             "How can we know how far away something is?",
-#             "How can we know how far away something is?",
-#             "How can we know how far away something is?",
-#             "How can we know how far away something is?",
-#             "Galaxy Distances",
-#             "Galaxy Distances",
-#             "Galaxy Distances",
-#             "Galaxy Distances",
-#             "Galaxy Distances",
-#             "Galaxy Distances",
-#             "Galaxy Distances"
-#         ],
-#         interact_steps=[7,9],
-#         distance_const=DISTANCE_CONSTANT,
-#         image_location=f"{IMAGE_BASE_URL}/stage_two_intro",
-#         debug = LOCAL_STATE.debug_mode.value,
-#     )
+
+import solara
+
+from hubbleds.components import Stage2Slideshow
+from hubbleds.state import LOCAL_STATE, GLOBAL_STATE 
+from .component_state import COMPONENT_STATE
+from hubbleds.remote import LOCAL_API
+from ...utils import IMAGE_BASE_URL, DISTANCE_CONSTANT
+
+from cosmicds.logger import setup_logger
+
+logger = setup_logger("STAGE 2")
+
+@solara.component
+def Page():
+    loaded_component_state = solara.use_reactive(False)
+
+    async def _load_component_state():
+        # Load stored component state from database, measurement data is
+        # considered higher-level and is loaded when the story starts
+        LOCAL_API.get_stage_state(GLOBAL_STATE, LOCAL_STATE, COMPONENT_STATE)
+
+        # TODO: What else to we need to do here?
+        logger.info("Finished loading component state for stage 2.")
+        loaded_component_state.set(True)
+
+    solara.lab.use_task(_load_component_state)
+
+    async def _write_component_state():
+        if not loaded_component_state.value:
+            return
+
+        # Listen for changes in the states and write them to the database
+        LOCAL_API.put_stage_state(GLOBAL_STATE, LOCAL_STATE, COMPONENT_STATE)
+
+        logger.info("Wrote component state for stage 2 to database.")
+
+    logger.info("Trying to write component state for stage 2.")
+    solara.lab.use_task(_write_component_state, dependencies=[COMPONENT_STATE.value])
+
+    Stage2Slideshow(
+        step = COMPONENT_STATE.value.distance_slideshow_state.step,
+        max_step_completed = COMPONENT_STATE.value.distance_slideshow_state.max_step_completed,
+        event_on_slideshow_finished=lambda _: COMPONENT_STATE.value.complete.set(True),
```

Comment:
There needs to be an `event_on_slideshow_finished` argument added to the `Stage2Slideshow` function, because currently this isn't passed in to the component.

Once that's added, this should be something like `Ref(COMPONENT_STATE.value.distance_slideshow_state.complete).set(True)`. The component state doesn't have a `complete` field so this would give an error (currently it doesn't execute because of the event not being passed in to the component).

Reply:
Just a quick note: will need to be `Ref(COMPONENT_STATE.fields.distance_slideshow_state.complete).set(True)` (note the `fields`).

Reply:
Yes, good catch, it should be `fields` rather than`value`